File: js/different_browsers.js
var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until,
    test = require('selenium-webdriver/testing');

test.describe('Firefox', function() {
    var driver;

    test.before(function() {
	driver = new webdriver.Builder()
	    .forBrowser('firefox')
	    .build();
    });

    test.it('should change title', function() {
	driver.get('https://duckduckgo.com');
	driver.wait(until.titleIs('DuckDuckGo'), 1000);
    });

    test.after(function() {
	driver.quit();
    });
});

test.describe('Chrome', function() {
    var driver;

    test.before(function() {
	driver = new webdriver.Builder()
	    .forBrowser('chrome')
	    .build();
    });

    test.it('should change title', function() {
	driver.get('https://duckduckgo.com');
	driver.wait(until.titleIs('DuckDuckGo'), 1000);
    });

    test.after(function() {
	driver.quit();
    });
});

File: ruby/different_browsers.rb
# coding: utf-8
require 'rspec'
require 'selenium-webdriver'

describe 'Chrome search' do
  before(:each) do
    @driver = Selenium::WebDriver.for :chrome
    @wait = Selenium::WebDriver::Wait.new(:timeout => 10)
  end

  it 'should find webdriver' do
    @driver.navigate.to 'http://duckduckgo.com'

    element = @driver.find_element(:name, 'q')
    @wait.until { @driver.title == 'DuckDuckGo'}
  end

  after(:each) do
    @driver.quit
  end
end

describe 'Firefox search' do
  before(:each) do
    @driver = Selenium::WebDriver.for :firefox
    @wait = Selenium::WebDriver::Wait.new(:timeout => 10)
  end

  it 'should find webdriver' do
    @driver.navigate.to 'http://duckduckgo.com'

    element = @driver.find_element(:name, 'q')
    @wait.until { @driver.title == 'DuckDuckGo'}
  end

  after(:each) do
    @driver.quit
  end
end

describe 'Internet Explorer search' do
  before(:each) do
    @driver = Selenium::WebDriver.for :internet_explorer
    @wait = Selenium::WebDriver::Wait.new(:timeout => 10)
  end

  it 'should find webdriver' do
    @driver.navigate.to 'http://duckduckgo.com'

    element = @driver.find_element(:name, 'q')
    @wait.until { @driver.title == 'DuckDuckGo'}
  end

  after(:each) do
    @driver.quit
  end
end
